src/Core/Helpers/PathHelper.cs

```diff
@@ -146,13 +146,61 @@ public static bool IsPathRootedUnderRoot(string path, string rootDirectory)
         public static bool IsMatchingPath(string path1, string path2) =>
             Path.GetFullPath(path1).Equals(Path.GetFullPath(path2), StringComparison.OrdinalIgnoreCase);
 
+        /// <summary>
+        /// Returns true if a local file path and a server file path should be considered to be equivalent, otherwise false
+        /// </summary>
+        /// <param name="absoluteLocalFilePath">The normalized local absolute file path. Can be null/empty.</param>
+        /// <param name="relativeServerFilePath">The normalized relative path from the server.  Can be null/empty.</param>
+        /// <remarks>
+        /// <paramref name="relativeServerFilePath"/> is expected to have been normalized by <see cref="SonarQube.Client.Helpers.FilePathNormalizer"/>.
+        /// Either parameter can be null/empty, in which case they are treated as a module (project) level issue. In that case
+        /// there will only be match if the other parameter is also a module level issue.
+        /// <para>
+        /// Implementation: the server path is tail-matched against the local path
+        /// e.g. the server path "aaa\foo.txt" will match ALL of the following absolute paths:
+        /// * C:\aaa\foo.txt
+        /// * C:\bbb\aaa\foo.txt
+        /// * D:\bbb\foo.txt
+        /// It will not match e.g. C:\XXXaaa\foo.txt
+        /// </para>
+        /// </remarks>
+        public static bool IsServerFileMatch(string absoluteLocalFilePath, string relativeServerFilePath)
```

Comment:
The implementation is the same, except I've changed the input parameters to be strings.
I added the doc comment.